polymode-compat.el

```diff
@@ -286,15 +286,18 @@ changes."
 
 (defun polymode-fix-desktop-buffer-info (fn buffer)
   "Save polymode buffers without mode prefix."
-  (let ((out (funcall fn buffer))
-        (base (buffer-base-buffer)))
+  (let* ((out (funcall fn buffer))
+         (base (buffer-base-buffer))
+         (base-name (and base
+                         (string-trim-left (buffer-name base)))))
     (with-current-buffer buffer
-      (if (not (and polymode-mode base))
-          out
+      (when (and polymode-mode base)
         (when (car out)
```

Comment:
I would prefer to use a nested `(when polymode-mode ... (when base ...))` instead of this sequential when.